assets/js/modules/analytics-4/components/widgets/TopTrafficSourceDrivingAddToCartWidget.js

```diff
@@ -0,0 +1,217 @@
+			MODULES_ANALYTICS_4
+		).getErrorForSelector( 'getReport', [ trafficSourceReportOptions ] );
+
+		const totalAddToCartReportErrors = select(
+			MODULES_ANALYTICS_4
+		).getErrorForSelector( 'getReport', [ totalAddToCartReportOptions ] );
+
+		if ( trafficSourceReportErrors && totalAddToCartReportErrors ) {
+			return [ trafficSourceReportErrors, totalAddToCartReportErrors ];
+		}
+
+		return (
+			trafficSourceReportErrors || totalAddToCartReportErrors || undefined
+		);
+	} );
+
+	const loading = useSelect( ( select ) =>
+		hasDetectedEvent
+			? ! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+					'getReport',
+					[ totalAddToCartReportOptions ]
+			  ) ||
+			  ! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+					'getReport',
+					[ trafficSourceReportOptions ]
+			  )
+			: undefined
+	);
+
+	const makeFilter = ( dateRange, dimensionIndex ) => ( row ) =>
+		get( row, `dimensionValues.${ dimensionIndex }.value` ) === dateRange;
+
+	// Prevents running a filter on `report.rows` which could be undefined.
+	const { rows: totalAddToCartReportRows = [] } = totalAddToCartReport || {};
+	const { rows: trafficSourceReportRows = [] } = trafficSourceReport || {};
+
+	const topTrafficSource =
+		trafficSourceReportRows.filter( makeFilter( 'date_range_0', 1 ) )[ 0 ]
+			?.dimensionValues?.[ 0 ].value || '-';
+
+	const currentTotalAddToCart =
+		parseInt(
+			totalAddToCartReportRows.filter(
+				makeFilter( 'date_range_0', 0 )
+			)[ 0 ]?.metricValues?.[ 0 ]?.value,
+			10
+		) || 0;
+
+	const currentTopTrafficSourceAddToCart =
+		parseInt(
+			trafficSourceReportRows.filter(
+				makeFilter( 'date_range_0', 1 )
+			)[ 0 ]?.metricValues?.[ 0 ]?.value,
+			10
+		) || 0;
+	const relativeCurrentTopTrafficSourceAddToCart = currentTotalAddToCart
+		? currentTopTrafficSourceAddToCart / currentTotalAddToCart
+		: 0;
+
+	const previousTotalAddToCart =
+		parseInt(
+			totalAddToCartReportRows.filter(
+				makeFilter( 'date_range_1', 0 )
+			)[ 0 ]?.metricValues?.[ 0 ]?.value,
+			10
+		) || 0;
+
+	const previousTopTrafficSourceAddToCart =
+		parseInt(
+			trafficSourceReportRows.filter(
+				makeFilter( 'date_range_1', 1 )
+			)[ 0 ]?.metricValues?.[ 0 ]?.value,
+			10
+		) || 0;
+	const relativePreviousTopTrafficSourceAddToCart = previousTotalAddToCart
+		? previousTopTrafficSourceAddToCart / previousTotalAddToCart
+		: 0;
+
+	const format = {
+		style: 'percent',
+		signDisplay: 'never',
+		maximumFractionDigits: 1,
+	};
+
+	return (
+		<MetricTileText
+			Widget={ Widget }
+			widgetSlug={ KM_ANALYTICS_TOP_TRAFFIC_SOURCE_DRIVING_ADD_TO_CART }
+			metricValue={ topTrafficSource }
+			metricValueFormat={ format }
+			subText={
+				// eslint-disable-next-line @wordpress/valid-sprintf
+				sprintf(
+					/* translators: %d: Percentage of leads for the current top traffic source compared to the number of total leads for all traffic sources. */
+					__( '%s of total add to carts', 'google-site-kit' ),
```

Comment:
You need to fix the comment to match translatable string placeholders.

Reply:
ah indeed, good catch